### PR TITLE
sdit-644 switching internal sync messages to use the synchronisation …

### PR DESCRIPTION
**Pull request by lbennett-moj**
…queue

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesPrisonOffenderEventListener.kt [hunk 1]
```diff
@@ -33,33 +34,50 @@ class IncentivesPrisonOffenderEventListener(
...
+        IncentiveMessages.RETRY_INCENTIVE_SYNCHRONISATION_MAPPING.name -> incentivesSynchronisationService.retryCreateIncentiveMapping(
+          sqsMessage.Message.fromJson()
+        )
       }
     }.thenAccept { }
   }
 
+  private inline fun <reified T> String.fromJson(): T =
+    objectMapper.readValue(this, object : TypeReference<T>() {})
+
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesPrisonOffenderEventListener.kt:80]: very minor, you shouldn't the second argument in `readValue`  assuming you import in the single argument Kotlin extension

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsMigrationServiceTest.kt [hunk 2]
```diff
@@ -381,24 +381,25 @@ internal class SentencingAdjustmentsMigrationServiceTest {
...
       @Test
-      internal fun `will check again in 10 second and reset even when previously started finishing up phase`(): Unit = runBlocking {
-        service.migrateStatusCheck(
-          MigrationContext(
-            type = SENTENCING_ADJUSTMENTS,
-            migrationId = "2020-05-23T11:30:00",
-            estimatedCount = 100_200,
-            body = MigrationStatusCheck(checkCount = 4)
+      internal fun `will check again in 10 second and reset even when previously started finishing up phase`(): Unit =
+        runBlocking {
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsMigrationServiceTest.kt:385]: how come you need to wrap all these in `runBlocking` what had changed?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsMigrationServiceTest.kt (hunk 2 above)
Comment (lbennett-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsMigrationServiceTest.kt:385]: looks like a mistake - will check

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**